processor/src/trace/utils.rs

```diff
@@ -237,6 +238,106 @@ impl HintCycle for u64 {
     }
 }
 
+// TRACE LENGTH SUMMARY
+// ================================================================================================
+
+/// Contains the data about lengths of the trace parts.
+///
+/// - `main_trace_len` contains the length of the main trace.
+/// - `range_trace_len` contains the length of the range checker trace.
+/// - `chiplets_trace_len` contains the trace lengths of the all chiplets (hash, bitwise, memory,
+/// kernel ROM)
+pub struct TraceLenSummary {
+    main_trace_len: usize,
+    range_trace_len: usize,
+    chiplets_trace_len: ChipletsLengths,
+}
+
+impl TraceLenSummary {
+    pub fn new(
+        main_trace_len: usize,
+        range_trace_len: usize,
+        chiplets_trace_len: ChipletsLengths,
+    ) -> Self {
+        TraceLenSummary {
+            main_trace_len,
+            range_trace_len,
+            chiplets_trace_len,
+        }
+    }
+
+    /// Returns length of the main trace
+    pub fn main_trace_len(&self) -> usize {
+        self.main_trace_len
+    }
+
+    /// Returns length of the range table
+    pub fn range_trace_len(&self) -> usize {
+        self.range_trace_len
+    }
+
+    /// Returns [ChipletsLengths] which contains trace lengths of all chilplets.
+    pub fn chiplets_trace_len(&self) -> ChipletsLengths {
+        self.chiplets_trace_len
+    }
+
+    /// Returns the maximum of all component lengths.
+    pub fn trace_len(&self) -> usize {
+        let chiplets_len = self.chiplets_trace_len.hash_chiplet_len()
+            + self.chiplets_trace_len.bitwise_chiplet_len()
+            + self.chiplets_trace_len.memory_chiplet_len()
+            + self.chiplets_trace_len.kernel_rom_len()
+            + 1;
```

Comment:
I would add `ChipletsLengths::trace_len()` method and move this computation there.